src/MoBi.Presentation/Presenter/EditFormulaPathListPresenter.cs

```diff
@@ -0,0 +1,249 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using MoBi.Assets;
+using MoBi.Core.Domain.Model;
+using MoBi.Core.Domain.Services;
+using MoBi.Core.Domain.UnitSystem;
+using MoBi.Core.Events;
+using MoBi.Presentation.DTO;
+using MoBi.Presentation.Mappers;
+using MoBi.Presentation.Presenter.BasePresenter;
+using MoBi.Presentation.Settings;
+using MoBi.Presentation.Views;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Domain.Formulas;
+using OSPSuite.Core.Domain.UnitSystem;
+using OSPSuite.Core.Extensions;
+using OSPSuite.Core.Services;
+using OSPSuite.Presentation.Core;
+using OSPSuite.Presentation.Presenters;
+using OSPSuite.Presentation.Presenters.ContextMenus;
+using OSPSuite.Utility.Events;
+using OSPSuite.Utility.Exceptions;
+
+namespace MoBi.Presentation.Presenter
+{
+   public interface IEditFormulaPathListPresenter :
+      IEditPresenter<IFormula>,
+      IPresenter<IEditFormulaPathListView>,
+      IPresenterWithContextMenu<IViewItem>,
+      IListener<FormulaChangedEvent>
+   {
+      IBuildingBlock BuildingBlock { set; }
+
+      /// <summary>
+      ///    Sets to <c>true</c> (default), circular reference check will be performed every time a path is added or edited.
+      /// </summary>
+      bool CheckCircularReference { get; set; }
+
+      /// <summary>
+      ///    Sets the alias of the <paramref name="formulaUsablePath" /> being edited to <paramref name="newAlias" /> from
+      ///    <paramref name="oldAlias" />
+      /// </summary>
+      void SetAlias(string newAlias, string oldAlias, IFormulaUsablePath formulaUsablePath);
+
+      void SetFormulaUsablePath(string newPath, FormulaUsablePathDTO dto);
+
+      /// <summary>
+      ///    Sets the formula dimension from <paramref name="oldValue" /> to <paramref name="newValue" />
+      /// </summary>
+      void SetFormulaPathDimension(IDimension newValue, IDimension oldValue, FormulaUsablePathDTO formulaUsablePathDTO);
+
+      void RemovePath(FormulaUsablePathDTO formulaUsablePathDTO);
+
+      IEnumerable<IDimension> GetDimensions();
+
+      /// <summary>
+      ///    Clones the path of <paramref name="formulaUsablePathToClone" /> and adds it to the formula with a unique alias
+      /// </summary>
+      void ClonePath(FormulaUsablePathDTO formulaUsablePathToClone);
+
+      /// <summary>
+      ///    Creates a new empty path in the formula
+      /// </summary>
+      void CreateNewPath();
+
+      Func<ReferenceDTO, bool> DragDropAllowedFor { get; set; }
+
+      void AddPathToFormula(IFormulaUsablePath path);
+
+      bool ReadOnly { set; get; }
+
+      void Drop(ReferenceDTO referenceDTO);
+
+      void Edit(IFormula formula, IUsingFormula formulaOwner);
+   }
+
+   public class EditFormulaPathListPresenter : AbstractEditPresenter<IEditFormulaPathListView, IEditFormulaPathListPresenter, IFormula>, IEditFormulaPathListPresenter
```

Comment:
THis is all the code from ExplicityFormulaParser taken out and into its own component so that I can reuse it for SumFormula